frontend/src/metabase-types/api/collection.ts

```diff
@@ -56,7 +56,8 @@ export interface Collection {
   personal_owner_id?: UserId;
   is_personal?: boolean;
 
-  location?: string;
+  location: string | null;
+  effective_location?: string; // location with only user-accessible collections
```

Comment:
unfortunately, not sure you can say a tiny bit here:

1. The metabase permissions model is ... unique ... in that a user can have access to a non-hierarchical set of collections
2. So there might be a set of collections `a/b/c/d` and a given user might only have access to collections `b` and `d`, so the effective path for `d` would be `b/d` for them because they can't access `a` or `c` 🤯 .